utils.py

```diff
@@ -205,6 +207,24 @@ def log_every(self, iterable, print_freq, header=None):
         print('{} Total time: {}'.format(header, total_time_str))
 
 
+def hex2rgb(hex: str):
+    return tuple(int(hex.lstrip('#')[i:i+2], 16) for i in (0, 2, 4))
+
+
+def create_colormap(labels: list) -> dict:
+    colormap = {0: (0, 0, 0)}
+    for i, label in enumerate(labels):
+        colormap[i+1] = hex2rgb(label['color'])
+    return colormap
+
+
+def create_palette(labels: dict):
+    palette = [[[0, 0, 0]]]
+    for label in labels:
+        palette.append([list(hex2rgb(label['color']))])
+    return Image.fromarray(np.uint8(np.array(palette))).convert("P")
```

Comment:
Creating the palette for training.
I created the dummy rgb image which reflects the training dataset to generate the color palette.